content/integrations/AWS-Integration/AWS-Detailed-Billing.md

```diff
@@ -30,10 +30,42 @@ weight: 4
 
 
 ## 2. Update your AWS integration in Metricly
+
+If you have just created your IAM role, **wait 2-5 minutes** for AWS to finalize its creation before proceeding to these steps. This ensures the new role has the correct s3 access permissions when added to Metricly.
+
 1. From the top navigation menu, select **Integrations**.
-2. Click the **Amazon Web Services** card.
+2. Select the **Amazon Web Services** card.
 3. Toggle **Detailed Billing** and scroll to the final section.
 ![Enable Detailed billing](/images/AWS-Detailed-Billing/enable-detailed-billing.png)
 4. Type the `S3 bucket name` into the corresponding field.
 The bucket name is case sensitive and must exactly match the bucket created in Step 1.  
-5. Click **Save**.  
+5. Select **Save**.  
+
+---
+
+
+## Detailed Billing Best Practices
+
+The following best practices ensure optimal detailed billing setup.
+
+### 1. Use Dedicated S3  Buckets
+Keep your billing files in one dedicated s3 bucket for each AWS account you have.
+
+**Doing so prevents**:
+
+- longer file queries
+- high processing cost
+- duplicate file names
+
+
+### 2. Request Detailed Billing Setup for Resold Resources
+
+You may not have access to your billing preferences if you have purchased AWS services through a reseller. You must contact your reseller and request to be set up with an s3 bucket to store your detailed billing data.
```

Comment:
Could you replace s3 with S3 (there are a few places). 

Not sure if this is technically an acronym but S3 is short for "Simple Storage Service"